app/routes.js

```diff
@@ -1,12 +1,18 @@
 const { healthCheck } = require('./controllers/healthCheck');
-const { createUserSchema, signInSchema, getUsersSchema } = require('./middlewares/validationSchemas/users');
+const {
+  createUserSchema,
+  signInSchema,
+  getUsersSchema,
+  upsertAdminSchema
+} = require('./middlewares/validationSchemas/users');
 const { validateSchema } = require('./middlewares/schemas');
-const { authUser } = require('./middlewares/authentication');
+const { authUser, authAdmin } = require('./middlewares/authentication');
 const usersController = require('./controllers/users');
 
 exports.init = app => {
   app.get('/health', healthCheck);
   app.post('/users', validateSchema(createUserSchema), usersController.createUser);
   app.post('/users/sessions', validateSchema(signInSchema), usersController.signIn);
   app.get('/users', authUser, validateSchema(getUsersSchema), usersController.getUsers);
+  app.post('/admin/users', authAdmin, validateSchema(upsertAdminSchema), usersController.upsertAdmin);
```

Comment:
you can add the validation in the authUser middle passing the role as a variable en this functions. Maybe something like: `authUser({ role: admin })` by default validate regular. users :)